packages/headless/src/features/search/search-actions.ts

```diff
@@ -113,6 +114,7 @@ export const prepareForSearchWithQuery = createAsyncThunk<
   });
 
   dispatch(deselectAllBreadcrumbs());
+  dispatch(updateFacetAutoSelection({allow: true}));
```

Comment:
Is there any other place where we should be dispatching that action ? I think that's only triggered whenever there's a query submit from the search box.

I think it's fine by the way, just asking the question out loud to make sure we dont forget any interaction.

Another approach might have been to have a different action than `deselectAllBreadcrumbs` when we dispatch `prepareForSearchWithQuery` : An action that removes all breadcrumb/deselect all facets, but that do not set prevent auto select (eg: `deselectAllBreadcrumbOnSearchboxSubmit` or whatever...) .

It is a little bit weird to have one action `deselectAllBreadcrumbs` that removes facets selection AND set the auto select flag to true, then right after another action that set the auto select flag to false.

It's not the end of the world, though...

Reply:
Yes we had the choice of making an action that is 90% similar to an existing one (with the flag being the only difference) or making a new action that can be used for possibly other use cases (we had nothing to allow or prevent autoselect directly).

Reply:
Yeah, there are multiple approaches, each with different pros and cons.

**Current approach**

Pro: flexibility, the action can be composed elsewhere, clear when reading.
Con: the order of dispatch is important, gives direct access to the property

**deselectAllBreadcrumbOnSearchboxSubmit**

Pro: single action, order independent.
Con: less flexible, not clear what the difference(s) is(are) with `deselectAllBreadcrumbs` (higher abstraction could be seen as a pro though)

**deselectAllBreadcrumbs({allowFacetAutoSelect: true})**

Pro: single action, order independent.
Con: possible to get it wrong, when should I set the param to true? (ah ... before a search, didn't know)


The current approach feels like a good balance,
but there are good arguments for taking alternative paths too